ui/src/FeastUISansProviders.tsx

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+
+import "@elastic/eui/dist/eui_theme_light.css";
+import "./index.css";
+
+import { Routes, Route } from "react-router-dom";
+import { EuiProvider, EuiErrorBoundary } from "@elastic/eui";
+
+import ProjectOverviewPage from "./pages/ProjectOverviewPage";
+import Layout from "./pages/Layout";
+import NoMatch from "./pages/NoMatch";
+import DatasourceIndex from "./pages/data-sources/Index";
+import DatasetIndex from "./pages/saved-data-sets/Index";
+import EntityIndex from "./pages/entities/Index";
+import EntityInstance from "./pages/entities/EntityInstance";
+import FeatureServiceIndex from "./pages/feature-services/Index";
+import FeatureViewIndex from "./pages/feature-views/Index";
+import FeatureViewInstance from "./pages/feature-views/FeatureViewInstance";
+import FeatureServiceInstance from "./pages/feature-services/FeatureServiceInstance";
+import DataSourceInstance from "./pages/data-sources/DataSourceInstance";
+import RootProjectSelectionPage from "./pages/RootProjectSelectionPage";
+import DatasetInstance from "./pages/saved-data-sets/DatasetInstance";
+import NoProjectGuard from "./components/NoProjectGuard";
+
+import TabsRegistryContext, {
+  FeastTabsRegistryInterface,
+} from "./custom-tabs/TabsRegistryContext";
+import FeatureFlagsContext, {
+  FeatureFlags,
+} from "./contexts/FeatureFlagsContext";
+import {
+  ProjectListContext,
+  ProjectsListContextInterface,
+} from "./contexts/ProjectListContext";
+
+interface FeastUIConfigs {
+  tabsRegistry: FeastTabsRegistryInterface;
+  featureFlags?: FeatureFlags;
+  projectListPromise?: Promise<any>;
+}
+
+const defaultProjectListPromise = () => {
+  return fetch("/projects-list.json", {
```

Comment:
seems more ideal to actually expose this as a parameter they can provide? 

e.g. a user may want this list to not exist locally and instead remotely on e.g. S3